utils/class-users-query-utils.php

```diff
@@ -30,4 +30,160 @@ public static function fix_found_users_query( $sql, $query ) {
+			$sql = "SELECT COUNT(DISTINCT {$wpdb->users}.ID) {$temp_query->query_from} {$temp_query->query_where} AND ({$capability_where})";
+			// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared,WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+			$result = $wpdb->get_var( $sql );
+			return (int) $result;
+		} else {
+			// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- SQL is built by WP_User_Query and internal methods
+			// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users -- Required for network-wide user capability filtering
+			$sql = "SELECT DISTINCT {$wpdb->users}.ID {$temp_query->query_from} {$temp_query->query_where} AND ({$capability_where})";
+			// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared,WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+			$results = $wpdb->get_col( $sql );
+			return array_map( 'intval', $results );
+		}
+	}
+
+	/**
+	 * Build a WHERE clause for network-wide capability/role filtering.
+	 *
+	 * This creates a subquery that checks if the user has the required
+	 * capabilities or roles on any site in the network using pattern matching.
+	 *
+	 * @param array $capabilities Required capabilities (OR logic).
+	 * @param array $roles        Required roles (OR logic).
+	 * @return string SQL WHERE clause condition (without WHERE keyword).
+	 */
+	private static function build_network_capability_where_clause( $capabilities, $roles ) {
+		global $wpdb;
+
+		$capabilities = array_filter( (array) $capabilities );
+		$roles        = array_filter( (array) $roles );
+
+		if ( empty( $capabilities ) && empty( $roles ) ) {
+			return '';
+		}
+
+		$capability_checks = [];
+
+		// Build capability conditions - convert capabilities to roles
+		if ( ! empty( $capabilities ) ) {
+			foreach ( $capabilities as $capability ) {
+				$roles_with_capability = self::get_roles_with_capability( $capability );
+				foreach ( $roles_with_capability as $role ) {
+					$role                = esc_sql( $role );
+					$capability_checks[] = "meta_value LIKE '%\"{$role}\";b:1%'";
+				}
+			}
+		}
+
+		// Build role conditions
+		if ( ! empty( $roles ) ) {
+			foreach ( $roles as $role ) {
+				$role                = esc_sql( $role );
+				$capability_checks[] = "meta_value LIKE '%\"{$role}\";b:1%'";
+			}
+		}
+
+		// Remove duplicates to avoid redundant SQL conditions
+		$capability_checks = array_unique( $capability_checks );
+
+		if ( empty( $capability_checks ) ) {
+			return '';
+		}
+
+		// Meta keys follow pattern: {prefix}_capabilities or {prefix}_{site_id}_capabilities
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users -- Required for network-wide user capability filtering
+		$subquery = "{$wpdb->users}.ID IN (
+			SELECT DISTINCT user_id
+			FROM {$wpdb->usermeta}
+			WHERE meta_key LIKE 'wp%_capabilities'
```

Comment:
nitpick: Might want to consider using the `base_prefix` to be future proof in case we'll have a non-default config

```
global $wpdb;
$base_prefix = $wpdb->base_prefix; // Always 'wp_' unless customized
```

Reply:
Good idea. Done in https://github.com/Automattic/vip-security-boost/pull/75/commits/41ad4633850a8c81d25fa12f59a9eb42b7297b87